client/src/lib.rs

```diff
@@ -0,0 +1,55 @@
+use fuel_debugger::{Command, Response};
+use std::io;
+use std::net::SocketAddr;
+use tokio::io::{AsyncBufReadExt, AsyncWriteExt, BufReader};
+use tokio::net::{
+    tcp::{OwnedReadHalf, OwnedWriteHalf},
+    TcpListener,
+};
+
+pub mod names;
+
+pub struct Listener {
+    listener: TcpListener,
+}
+
+impl Listener {
+    pub fn new(listener: TcpListener) -> Self {
+        Self { listener }
+    }
+
+    pub async fn accept(&self) -> io::Result<(Client, SocketAddr)> {
+        let (socket, addr) = self.listener.accept().await?;
+
+        let (reader, writer) = socket.into_split();
+        let reader = BufReader::new(reader);
+
+        Ok((Client { reader, writer }, addr))
+    }
+}
+
+pub struct Client {
+    reader: BufReader<OwnedReadHalf>,
+    writer: OwnedWriteHalf,
+}
+
+impl Client {
+    pub async fn cmd(&mut self, cmd: &Command) -> io::Result<Response> {
+        let mut v = serde_json::to_string(cmd).expect("Serialization failed");
```

Comment:
Do we really want to panic here when we could trivially return an `io::Error`? Also, below, we return an error in case the response isn't appropriate. I understand we might want to panic in case of CLIs, but then we are restricting this object for its own usage instead of making it available as library.